wandb/apis/public/registries/_members.py

```diff
@@ -0,0 +1,110 @@
+"""Types and helpers for managing registry members."""
+
+from __future__ import annotations
+
+from collections import defaultdict
+from enum import Enum
+from functools import singledispatchmethod
+from typing import Iterable, Literal, Union
+
+from pydantic.dataclasses import dataclass as pydantic_dataclass
+
+from wandb._strutils import b64decodestr, b64encodestr, nameof
+from wandb.sdk.artifacts._models import ArtifactsBase
+
+from ..teams import Team
+from ..users import User
+
+
+class MemberKind(str, Enum):
+    """Identifies what kind of object a registry member is."""
+
+    USER = "User"
+    ENTITY = "Entity"
+
+    TEAM = ENTITY  # Convenience alias
+
+
+class MemberRole(str, Enum):
```

Comment:
Will we be able to expand this in the future if we add support for custom roles?

Reply:
This plan sgtm! Just wanted to double check. I think its okay to leave the tests as is since there's no immediate plan to bring in custom roles (AFAIK).